### PR TITLE
fix: Upgrade broken on VMSS cluster after cluster-autoscaler scale out 

### PR DESCRIPTION
**Pull request by chengliangli0918**
**Reason for Change**:
upgrade broken on VMSS cluster after cluster-autoscaler scale out (https://github.com/Azure/aks-engine/issues/401)

**Issue Fixed**:
upgrade broken on VMSS cluster after cluster-autoscaler scale out (https://github.com/Azure/aks-engine/issues/401)

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
This is great, thanks @chengliangli0918 ! Is there anything that should be updated in the documentation regarding this change? https://github.com/Azure/aks-engine/blob/master/docs/topics/upgrade.md#L95

**Comment by chengliangli0918** [comment]
> This is great, thanks @chengliangli0918 ! Is there anything that should be updated in the documentation regarding this change? https://github.com/Azure/aks-engine/blob/master/docs/topics/upgrade.md#L95

I don't think the upgrade.md doc needs update.  Not all internal logic details in upgrader.go are documented in upgrade.md. For example, the step ARM template deployment in upgradeMasterNodes() (which causes this issue https://github.com/Azure/aks-engine/issues/401) is not mentioned in upgrade.md

**Comment by CecileRobertMichon** [comment]
@chengliangli0918 I was referring to this part specifically https://github.com/Azure/aks-engine/blob/master/docs/topics/upgrade.md#cluster-autoscaler--vmss

File: pkg/operations/kubernetesupgrade/upgrader.go [hunk 1]
```diff
@@ -80,6 +86,44 @@ func (ku *Upgrader) Validate() error {
...
+func (ku *Upgrader) setActualNodeCountForVMSSAgentPool() map[string]int {
+	nodeCountInProfile := make(map[string]int)
+	agentPoolScaleSetsToUpgradeMap := make(map[string]AgentPoolScaleSet)
+
+	for _, vmssToUpgrade := range ku.ClusterTopology.AgentPoolScaleSetsToUpgrade {
+		poolName, _, _ := utils.VmssNameParts(vmssToUpgrade.Name)
+		agentPoolScaleSetsToUpgradeMap[poolName] = vmssToUpgrade
+	}
+
+	// If the agent pool is VMSS agent pool, the actual number of node in VMSS might be different from the node count in the profile.
```
Comment (feiskyer) [inline pkg/operations/kubernetesupgrade/upgrader.go]: If the vmss is under scaling, though the capacity value is right, the actual virtual machine count may be still different. Do we need to wait for scaling finish?

**Comment by feiskyer** [comment]
Since this has been closed, @jackfrancis would you fix this in another way for aks-engine users?

**Comment by CecileRobertMichon** [comment]
@chengliangli0918 why did you close the PR?

**Comment by chengliangli0918** [comment]
@CecileRobertMichon: Yesterday in an internal email thread (@jackfrancis is on that thread), AKS team decided to use another approach at AKS side to fix  issue 401 that also happens on VMSS + CAS AKS public preview. So I abandoned this aks-engine PR.

If aks-engine team think this PR is ok for the issue 401, I can reopen/recreate another PR with this change.

Another thing I'd like to call out, on AKS VMSS + CAS preview, the CAS is temporarily disabled during upgrading. Not sure if aks-engine disables CAS during upgrading? This PR should fix issue 401. But if CAS is still active during upgrading on aks-engine cluster, CAS might cause other potential issue during upgrading.

> @chengliangli0918 why did you close the PR?

**Pull request closed without merging**